docs/firmware/payload_uboot.md

```diff
@@ -0,0 +1,32 @@
+U-Boot as a payload to OpenSBI
+==============================
+
+U-Boot is an open-source primary boot loader. It can be used as first and/or
+second stage boot loader in an embedded environment. Here, it is used as next
+stage bootloader to OpenSBI if given as a payload to OpenSBI.
+
+Currently, some of the required patches are not mainline. Thus, you need to
+cherry-pick the following out-of-tree patch series until it is accepted in mainline.
+
+HiFive Unleashed support for U-Boot
+
+https://lists.denx.de/pipermail/u-boot/2019-February/358058.html
+
+This patch series enables only 1 cpu to enter to U-Boot. As a result, next stage
+supervisor OS such as Linux can also boot only 1 cpu. SMP support in U-Boot
+for RISC-V can be enabled with following additional patches here.
+
+https://lists.denx.de/pipermail/u-boot/2019-February/358393.html
+
+Building and Generating U-Boot images
+=====================================
+Please refer U-Boot building documentation on steps how to build U-Boot images.
```

Comment:
Please refer to U-Boot build documentation for detailed instructions on how to build U-Boot images.